system-parachains/asset-hubs/asset-hub-kusama/src/treasury.rs

```diff
@@ -0,0 +1,189 @@
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot. If not, see <http://www.gnu.org/licenses/>.
+
+//! This file contains relevant configuration of treasury (migrated from the RC with AHM).
+
+use super::*;
+
+use crate::governance::{Treasurer, TreasurySpender};
+
+use frame_support::traits::tokens::{Pay, PaymentStatus, UnityAssetBalanceConversion};
+use frame_support::traits::{Currency, Get, OnUnbalanced};
+use scale_info::TypeInfo;
+use sp_runtime::traits::IdentityLookup;
+use sp_runtime::DispatchError;
+use xcm::prelude::*;
+
+parameter_types! {
+	pub const TreasuryPalletId: PalletId = PalletId(*b"py/trsry");
+	pub const MaxApprovals: u32 = 100;
+
+	// TODO: AH or RC DAYS?
+	pub const SpendPeriod: BlockNumber = 6 * DAYS;
+	pub const PayoutSpendPeriod: BlockNumber = 90 * DAYS;
+
+	// TODO: revisit !!! Location on RC, find out how is
+	// The asset's interior location for the paying account. This is the Treasury
+	// pallet instance (which sits at index 18).
+	// pub TreasuryInteriorLocation: InteriorLocation = PalletInstance(TREASURY_PALLET_ID).into();
+}
+
+#[derive(
+	Default,
+	MaxEncodedLen,
+	Encode,
+	Decode,
+	DecodeWithMemTracking,
+	TypeInfo,
+	Clone,
+	Eq,
+	PartialEq,
+	Debug,
+)]
+pub struct BurnDestinationAccount(pub Option<polkadot_core_primitives::AccountId>);
+
+impl BurnDestinationAccount {
+	pub fn is_set(&self) -> bool {
+		self.0.is_some()
+	}
+}
+
+pub type BalancesNegativeImbalance = <Balances as Currency<AccountId>>::NegativeImbalance;
+pub struct TreasuryBurnHandler;
+
+impl OnUnbalanced<BalancesNegativeImbalance> for TreasuryBurnHandler {
+	fn on_nonzero_unbalanced(amount: BalancesNegativeImbalance) {
+		let destination = dynamic_params::treasury::BurnDestination::get();
+
+		if let BurnDestinationAccount(Some(account)) = destination {
+			// Must resolve into existing but better to be safe.
+			Balances::resolve_creating(&account, amount);
+		} else {
+			// If no account to destinate the funds to, just drop the
+			// imbalance.
+			<() as OnUnbalanced<_>>::on_nonzero_unbalanced(amount)
+		}
+	}
+}
+
+impl Get<Permill> for TreasuryBurnHandler {
+	fn get() -> Permill {
+		let destination = dynamic_params::treasury::BurnDestination::get();
+
+		if destination.is_set() {
+			dynamic_params::treasury::BurnPortion::get()
+		} else {
+			Permill::zero()
+		}
+	}
+}
+
+impl pallet_treasury::Config for Runtime {
+	type PalletId = TreasuryPalletId;
+	type Currency = Balances;
+	type RejectOrigin = EitherOfDiverse<EnsureRoot<AccountId>, Treasurer>;
+	type RuntimeEvent = RuntimeEvent;
+	type SpendPeriod = SpendPeriod;
+	type Burn = TreasuryBurnHandler;
+	type BurnDestination = TreasuryBurnHandler;
+	type MaxApprovals = MaxApprovals;
+	type WeightInfo = weights::pallet_treasury::WeightInfo<Runtime>;
+	type SpendFunds = Bounties;
+	type SpendOrigin = TreasurySpender;
+
+	// TODO: Do we still need `VersionedLocatableAsset`? (Check treasury migration!)
+	type AssetKind = VersionedAssetId;
```

Comment:
@muharem I checked how the treasury is migrated and also: https://github.com/polkadot-fellows/runtimes/blob/dev-asset-hub-migration/pallets/rc-migrator/src/treasury.md
With support only for `LocalPay` and `Location { parents: 0, interior: Here }`, can we just use `VersionedAssetId` and `VersionedLocation` instead of `VersionedLocatableAsset` and `VersionedLocatableAccount`?

Instead of this:
```
type AssetKind = VersionedLocatableAsset;
type Beneficiary = VersionedLocatableAccount;
```
to use:
```
type AssetKind = VersionedAssetId;
type Beneficiary = VersionedLocation;
```